replication/transaction_payload_event.go

```diff
@@ -0,0 +1,149 @@
+package replication
+
+import (
+	"encoding/binary"
+	"encoding/hex"
+	"fmt"
+	"io"
+
+	"github.com/DataDog/zstd"
```

Comment:
nit: Have you compared this package with https://github.com/klauspost/compress? :) 
There is (an old) [comparison](https://github.com/klauspost/compress/issues/109) for some zstd implementations.